models/user.ts

```diff
@@ -3,3 +3,14 @@ export interface User {
   email: string;
   workspaces: string[];
 }
+
+export interface UserResponse extends Omit<User, 'workspaces'> {
+  workspaces: UserWorkspace[];
+}
+
+export interface UserWorkspace {
+  id: string;
+  workspaceName: string;
+  users: string[];
+  formCount: number;
+}
```

Comment:
I think our models are getting a bit confusing, especially because it seems we're trying to share them between the client and server. In the components I see where doing a union with `WorkspaceSnapshot`.

Thought maybe you might have some ideas on the models here, I think we should address them before they get out of hand.

Reply:
I just didn't think we're at the point of making view models and models yet. Would prefer to keep them in one place until we have a reason for interfacing around them.

Reply:
That being said if we want a really clear separation, we just need to do an equivalent viewmodel for everything coming back from the backend

Reply:
I'm going to pull this down and see if I can get a better feel for how bad this is

Reply:
so furthermore, user is always going to be an outlier because it's not an Entity, so may just deserve a little special treatment

Reply:
Okay, so looking at this, I wouldn't think any of these need to be added. Why can't you just user `WorkspaceSnapshot[]` ? It's all the same fields, these just duplicate them

Reply:
It extended the snapshot base which had a bunch of fields I don’t think I needed, especially for optimistic response, maybe I could do a Pick<> on it, but it was a late night at 2am, maybe I was going crazy

Reply:
check out my proposal PR

Reply:
I looked at it and commented, but dont think its needed anymore with codegen since that is now providing always up to date accurate types based on our schema. We might even wamt to leverage it on the server, there is a plugin for resolvers, etc that integrates with apollo server.